zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Note.java

```diff
@@ -655,6 +655,13 @@ public void runAll(AuthenticationInfo authenticationInfo, boolean blocking) {
       }
       if (authenticationInfo != null) {
         p.setAuthenticationInfo(authenticationInfo);
+      } else if (p.getUser() != null){
+        // set AuthenticationInfo of this paragraph using its user when authenticationInfo is null.
+        // authenticationInfo will be null if cronjob started before user run this
+        // paragraph manually.
+        p.setAuthenticationInfo(new AuthenticationInfo((p.getUser())));
+      } else {
+        p.setAuthenticationInfo(AuthenticationInfo.ANONYMOUS);
```

Comment:
looks ok, is it better to make sure authenticationInfo is set when scheduling the cronjob?